### PR TITLE
CBL-5795: Explicitly close FLSliceResults

### PR DESCRIPTION
**Pull request by bmeike**
All FLSliceResults are now explicitly closed after use.

### CODE REVIEW

File: common/main/java/com/couchbase/lite/AbstractQuery.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public void setParameters(@Nullable Parameters parameters) throws CouchbaseLiteE
 
             if (parameters == null) { return; }
 
-            try { getC4QueryLocked().setParameters(parameters.encode()); }
+            try (FLSliceResult encodedParams = parameters.encode()) { getC4QueryLocked().setParameters(encodedParams); }
```
Comment (pasin) [inline common/main/java/com/couchbase/lite/AbstractQuery.java:140]: From the code, I couldn't tell whether setParameters will copy bytes or retain the FLSliceResult or not. Can you confirm if it is doing one of the above?

File: common/main/java/com/couchbase/lite/AbstractDatabase.java [hunk 2]
```diff
@@ -1497,12 +1497,13 @@ private void saveResolvedDocumentWithFlags(
...
-                final FLSliceResult mergedBody = resolvedDoc.encode();
-                // if the resolved doc has attachments, be sure has the flag
-                if (C4Document.dictContainsBlobs(mergedBody, sharedKeys)) {
-                    mergedFlags |= C4Constants.RevisionFlags.HAS_ATTACHMENTS;
+                try (FLSliceResult mergedBody = resolvedDoc.encode()) {
+                    // if the resolved doc has attachments, be sure it has the flag
+                    if (C4Document.dictContainsBlobs(mergedBody, sharedKeys)) {
+                        mergedFlags |= C4Constants.RevisionFlags.HAS_ATTACHMENTS;
+                    }
+                    mergedBodyBytes = mergedBody.getContent();
```
Comment (pasin) [inline common/main/java/com/couchbase/lite/AbstractDatabase.java:1505]: Expect that getContent() will copy the bytes?

File: common/main/java/com/couchbase/lite/AbstractQuery.java (hunk 1 above)
Comment (bmeike) [reply common/main/java/com/couchbase/lite/AbstractQuery.java:140]: `setParameters` is a call to `c4query_setParameters`, which takes as an argument, a `C4String`.  According to @snej , there is no need to worry about C4String parameters, after a call returns.

File: common/main/java/com/couchbase/lite/AbstractDatabase.java (hunk 2 above)
Comment (bmeike) [reply common/main/java/com/couchbase/lite/AbstractDatabase.java:1505]: It does.

**Review by pasin — APPROVED** [review]

**Pull request merged**